kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -392,11 +400,11 @@ def intlist(ints: list[KInner]) -> KApply:
         return res
 
     @staticmethod
-    def typed_args(args: list[KInner]) -> KApply:
-        res = KApply('.List{"_,__EVM-ABI_TypedArgs_TypedArg_TypedArgs"}_TypedArgs')
-        for i in reversed(args):
-            res = KApply('_,__EVM-ABI_TypedArgs_TypedArg_TypedArgs', [i, res])
-        return res
+    def typed_args(args: list[KApply], res: KApply | None = None) -> KApply:
```

Comment:
Unless there's reason to expose the unit, I'd just go with the default:

```suggestion
    def typed_args(args: list[KApply]) -> KApply:
```